src/libs/ajax/GoogleStorage.ts

```diff
@@ -159,13 +181,29 @@ export const GoogleStorage = (signal?: AbortSignal) => ({
     return _.filter(({ name }) => _.includes(getExtension(name), runtimeTools.Jupyter.ext), items)
   },
 
-  listAnalyses: async (googleProject, name) => {
+  listAnalyses: async (googleProject: string, name: string): Promise<AnalysisFile[]> => {
     const res = await fetchBuckets(
       `storage/v1/b/${name}/o?prefix=notebooks/`,
       _.merge(authOpts(await saToken(googleProject)), { signal })
     )
-    const { items } = await res.json()
-    return _.filter(({ name }) => (_.includes(getExtension(name), runtimeTools.Jupyter.ext) || _.includes(getExtension(name), runtimeTools.RStudio.ext)), items)
+
+    const { items } = await res.json() as GoogleStorageListRaw
+    const internalFiles = _.flow(
+      _.map(({ name, updated, metadata }) => {
+        const path = name as AbsolutePath
+        return {
+          name: path,
+          ext: getExtension(name),
+          displayName: getDisplayName(name),
+          fileName: getFileName(name),
+          tool: getToolFromFileExtension(getExtension(name)) as ToolLabel,
+          lastModified: new Date(updated).getTime(),
+          cloudProvider: cloudProviderTypes.GCP,
+          metadata
+        }
+      }),
+      _.filter(({ ext }) => (_.includes(ext, runtimeTools.Jupyter.ext) || _.includes(ext, runtimeTools.RStudio.ext))))(items)
```

Comment:
Formatting nit to more clearly separate the last step in the flow from the input.

```suggestion
      _.filter(({ ext }) => (_.includes(ext, runtimeTools.Jupyter.ext) || _.includes(ext, runtimeTools.RStudio.ext)))
    )(items)
```